kernel/packages/unity-interface/BrowserInterface.ts

```diff
@@ -178,11 +183,25 @@ export class BrowserInterface {
     globalThis.globalStore.dispatch(saveProfileRequest({ ...profile, interests: Array.from(unique) }))
   }
 
-  public SaveUserAvatar(changes: { face: string; face128: string; face256: string; body: string; avatar: Avatar }) {
+  public SaveUserAvatar(changes: {
+    face: string
+    face128: string
+    face256: string
+    body: string
+    avatar: Avatar
+    isSignUpFlow?: boolean
+  }) {
     const { face, face128, face256, body, avatar } = changes
     const profile: Profile = getUserProfile().profile as Profile
     const updated = { ...profile, avatar: { ...avatar, snapshots: { face, face128, face256, body } } }
-    globalThis.globalStore.dispatch(saveProfileRequest(updated))
+    if (!changes.isSignUpFlow) {
+      globalThis.globalStore.dispatch(saveProfileRequest(updated))
+    } else {
+      globalThis.globalStore.dispatch(signUpSetProfile(updated))
+      globalThis.globalStore.dispatch(changeSignUpStage('passport'))
+      unityInterface.DeactivateRendering()
+      document.getElementById('gameContainer')!.setAttribute('style', 'display: none')
```

Comment:
Can't you put this in the `HTML` module that you've created?

Reply:
yes! of course, I've seen that line duplicated in so many places